### PR TITLE
Add GNMI client cert cname check support.

### PR DESCRIPTION
**Pull request by liuh-80**
Add GNMI client cert cname list to yang model.

#### Why I did it
Allow gnmi service authentication client cert by cname.

##### Work item tracking
- Microsoft ADO: 25226269

#### How I did it
Add GNMI client cert cname list to yang model.

#### How to verify it
Pass all UT.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

will updated with this PR image later.
- [] SONiC.master-16482.360728-2c8b4066f

#### Description for the changelog
Add GNMI client cert cname list to yang model.

### CODE REVIEW

**Comment by liuh-80** [comment]
Submodule PR: https://github.com/sonic-net/sonic-gnmi/pull/241

File: src/sonic-yang-models/yang-models/sonic-telemetry.yang [hunk 1]
```diff
@@ -74,5 +74,26 @@ module sonic-telemetry {
...
+                        "client cert common name";
+                }
+
+                leaf role {
+                    type string;
+                    description
+                        "role of client cert common name";
+                }
+            }
+        }
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-telemetry.yang:97]: Please update unit test for sonic-yang-models

Comment (liuh-80) [reply src/sonic-yang-models/yang-models/sonic-telemetry.yang:99]: Fixed

File: src/sonic-yang-models/yang-models/sonic-telemetry.yang [hunk 2]
```diff
@@ -74,5 +74,27 @@ module sonic-telemetry {
             }
 
         }
+
+        container TELEMETRY_CLIENT_CERT {
+            description "TELEMETRY client cert list";
+
+            list TELEMETRY_CLIENT_CERT_LIST {
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-telemetry.yang:81]: >TELEMETRY_CLIENT_CERT_LIST [](http://example.com/codeflow?start=17&length=26)

similar to GNMI_CLIENT_CERT_LIST. Can we reuse?

Comment (liuh-80) [reply src/sonic-yang-models/yang-models/sonic-telemetry.yang]: Fixed, reuse GNMI_CLIENT_CERT table for both GNMI and Telemetry service

**Review by qiluo-msft — APPROVED** [review]

**Comment by liuh-80** [comment]
Manually cherry-pick not found conflict: https://github.com/sonic-net/sonic-buildimage/pull/20792

**Comment by liuh-80** [comment]
Here is test case PR, also need cherry-pick later: https://github.com/sonic-net/sonic-mgmt/pull/13133

**Comment by r12f** [comment]
hi @FengPan-Frank , the change is conflicting with 202412, I wonder if you could help with a manual cherry pick?

**Pull request merged**